### PR TITLE
Metadata: list of services questions to scan for bad words

### PR DESCRIPTION
**Pull request by katstevens**
Trello: https://trello.com/c/Q0tQI9xL/439-spike-scan-g-cloud-services-for-bad-words-script-can-we-run-this-against-draft-services

Creates a new metadata list of service questions with free text inputs, that we will need to scan for bad/non-compliant words.

The `scan-g-cloud-services-for-bad-words` script (see https://github.com/alphagov/digitalmarketplace-scripts/blob/master/scripts/scan-g-cloud-services-for-bad-words.py) currently uses a hardcoded list of service questions from all recent frameworks (G7-G10). The script only ever scans services for one framework at a time, however, so we might as well start being framework specific now. See https://github.com/alphagov/digitalmarketplace-frameworks/tree/master/frameworks/g-cloud-11/questions/services for a list of G11 service questions.

The script itself will be refactored to use this list separately.

### CODE REVIEW

**Comment by risicle** [comment]
'td be nice if we could have the tests assert that all of these field names actually exist.

**Comment by katstevens** [comment]
@risicle that is an excellent idea.

**Comment by katstevens** [comment]
Test now added, along with some schema validation (and a typo fix that the test caught!)

**Comment by lfdebrux** [comment]
> Creates a new metadata list of service questions with free text inputs, that we will need to scan for bad/non-compliant words.

Did you create this file by hand or using a script? It would be good if that information were in the commit message?

File: tests/test_metadata.py [hunk 1]
```diff
@@ -32,3 +32,33 @@ def test_copying_services_metadata():
...
+                    continue
+                with open(f'frameworks/g-cloud-11/questions/services/{filename}') as f:
+                    contents = yaml.safe_load(f)
+                    if contents.get('id'):
+                        names_ids_fields.add(contents['id'])
+                    if contents.get('fields'):
+                        names_ids_fields.update([v for v in contents.get('fields').values()])
+                    names_ids_fields.add(filename[:-4])
+
+            for question in copying_services_data['service_questions']:
```
Comment (lfdebrux) [inline tests/test_metadata.py]: Is this for loop meant to be in the `for service_question_filename` loop?

Comment (katstevens) [reply tests/test_metadata.py]: Some c+p from the test above has sneaked in. Will push a commit to fix.

**Comment by katstevens** [comment]
@lfdebrux updated with the suggestions, thanks! I'll update the manual for this once the script's merged.

**Pull request merged**